client/signup/steps/domain-upsell/index.jsx

```diff
@@ -45,8 +51,17 @@ export default function DomainUpsellStep( props ) {
 		}
 	);
 
+	const isInDomainUpsellEmphasizeFreeTest =
+		'treatment' ===
+		useSelector( ( state ) => getVariationForUser( state, 'domain_upsell_emphasize_free_v2' ) );
+
 	return (
-		<div className="domain-upsell__step-secton-wrapper">
+		<div
+			className={ classnames( 'domain-upsell__step-secton-wrapper', {
+				is_in_domain_upsell_emphasize_free_test: isInDomainUpsellEmphasizeFreeTest,
+			} ) }
+		>
+			<Experiment name="domain_upsell_emphasize_free_v2" />
```

Comment:
This component actually does the requesting for the experiment, but you select it on line 54, before the state is actually set. So, you may have a treatment, but due to the race condition implemented here, you'd pass `false`. I'd recommend taking advantage of the component:

```suggestion
			<Experiment name="domain_upsell_emphasize_free_v2">
			    <Control>
			      <StepWrapper stepContent={ RecommendedDomains( props ) } />
			    </Control>
			    <Variation name="treatment">
			      <StepWrapper stepContent={ RecommendedDomains( { ...props, isInDomainUpsellEmphasizeFreeTest: true  } ) }
			    </Variation>
			    <LoadingVariations>
			      <StepWrapper stepContent={ RecommendedDomains( { ...props, isExperimentLoading: true } ) }
			    </LoadingVariations>
			</Experiment>
```

cc: @jessie-ross just to validate it still works this way.

Reply:
Thanks! If I understand correctly, the `<Experiment/>` component requests all experiments and updates the redux state. If the experiment data is not already in the state, it should cause this component to re-render and use the new value of the assigned variation. The `RecommendedDomains` component does handle the loading state too. Can you please explain how a race condition would be generated here?

Reply:
Good call @withinboredom to double check this, it does look like `<Experiment>` should ideally be used here.

AFAICT it looks to be alright to me thanks to the ` && !loading ` in `RecommendedDomains` - so long as the added class doesn't affect anything until the loading has finished.

Reply:
The main concern is of course is displaying the control early - manually testing the loading state should rule this out.